### PR TITLE
amazon upsell link #121

### PR DESCRIPTION
**Pull request by contactashish13**

### CODE REVIEW

File: inc/wppr-filters.php [hunk 1]
```diff
@@ -84,3 +84,19 @@ function wppr_review_cons_text_filter( $id = 0, $name = '' ) {
 	return ( ! empty( $name ) )  ? '<h2>'.$name.'</h2>' : '';
 
 }
+
+
+add_filter( 'cwp_admin_pointers-post', 'cwp_admin_pointers' );
```
Comment (selul) [inline inc/wppr-filters.php]: use wppr instead of cwp prefx

File: inc/cwp_metabox.php [hunk 2]
```diff
@@ -123,6 +123,7 @@ function cwp_review_meta_boxes(){
                     }
 
                     ?>"/>
+                    <span id="cwp_product_affiliate_link_upsell" style="display: none">&nbsp;</span>
```
Comment (selul) [inline inc/cwp_metabox.php]: use wppr instead of cwp id prefix

File: inc/wppr-main.php [hunk 3]
```diff
@@ -344,7 +344,44 @@ function custom_bar_icon() {
 		wp_register_script("cwp-custom-bar-icon", WPPR_URL.'/javascript/custom-bar-icon.js', false, "1.0", "all");
 		wp_enqueue_script("cwp-custom-bar-icon");
 	}
+
+    cwp_add_pointers();
+}
+
+function cwp_add_pointers()
```
Comment (selul) [inline inc/wppr-main.php]: same here use wppr prefix

File: javascript/cwp-pointers.js [hunk 4]
```diff
@@ -0,0 +1,20 @@
+(function ($, cwpp){
+
+    $(document).ready( function() {
+
+        $("#cwp_product_affiliate_link").on("keyup", function(e){
+            if($(this).val().indexOf("amazon") != -1){
+                $("#cwp_product_affiliate_link_upsell").show();
```
Comment (selul) [inline javascript/cwp-pointers.js]: please add a timeout and not show right away, the timeout should be 500ms. be careful to clear it in the bind function, something like this 
```
clearTimeout (timeout);
setTimeout( function (){ show_pointer() },500 );

```

File: inc/wppr-main.php [hunk 5]
```diff
@@ -344,7 +344,44 @@ function custom_bar_icon() {
...
+{
+    $screen     = get_current_screen();
+    $screen_id  = $screen->id;
+     
+    // Get pointers for this screen
+    $pointers = apply_filters( 'cwp_admin_pointers-' . $screen_id, array() );
```
Comment (selul) [inline inc/wppr-main.php]: use wppr prefix to the filter

File: inc/wppr-main.php [hunk 6]
```diff
@@ -344,7 +344,44 @@ function custom_bar_icon() {
...
+        $valid['pointers'][] =  $pointer;
+    }
+ 
+    if ( empty( $valid ) ) return;
+ 
+    // Add pointers style to queue.
+    wp_enqueue_style( 'wp-pointer', array("jquery"));
+
+    wp_enqueue_script( 'cwp-pointers', WPPR_URL.'/javascript/cwp-pointers.js', array("wp-pointer"), WPPR_LITE_VERSION, true );
+    wp_localize_script( 'cwp-pointers', 'cwpp', array("pointers" => $valid) );
```
Comment (selul) [inline inc/wppr-main.php]: same here use wppr

File: javascript/cwp-pointers.js (hunk 4 above)
Comment (selul) [inline javascript/cwp-pointers.js]: @contactashish13 dont add wptuts prefix, use wppr.

Comment (selul) [inline javascript/cwp-pointers.js]: also make sure you dont show the popup anymore once they clicked dismiss

Comment (contactashish13) [inline javascript/cwp-pointers.js]: @selul dont show the pop up for this product or dont show the popup for this user?

Comment (selul) [inline javascript/cwp-pointers.js]: you can save the flag value inside user_meta, so for user

**Pull request merged**